pandas/core/algorithms.py

```diff
@@ -356,7 +356,7 @@ def unique(values):
     # categorical is a fast-path
     # this will coerce Categorical, CategoricalIndex,
     # and category dtypes Series to same return of Category
-    if is_categorical_dtype(values):
+    if is_extension_array_dtype(values):
         values = getattr(values, '.values', values)
```

Comment:
Ha, this `getattr` is doing nothing since it's `'.values'` and not `'values'` :)

Reply:
Ah, and we have a test that depends on it! I'm going to split this off to a new issue then.

Reply:
Ah, nevermind. I think we can just remove it instead of correcting it.